src/web/src/pages/room/VotingRoomPage/index.tsx

```diff
@@ -47,52 +51,144 @@ function QuestionVoter({ data }: { data: VotingPageState }) {
-        {candidatesReordered.map(candidate => (
-          <Button
-            className={twMerge(
-              lastVote?.type === 'SingleVote' && lastVote.candidateId === candidate.id && 'btn-accent',
+      <div className="flex gap-4 flex-wrap flex-col items-stretch justify-center w-full">
+        {question.details.type === 'SingleVote'
+          ? question.candidates.map(candidate => (
+            <Controller
+              name="candidateId"
+              control={control}
+              key={candidate.id}
+              render={({ field: { value, onChange } }) => (
+                <Button
+                  className={twMerge(
+                    value === candidate.id ? 'btn-accent' : undefined,
+                  )}
+                  onClick={() => {
+                    setValue('type', 'SingleVote');
+                    onChange(candidate.id);
+                    onSubmit();
+                  }}
+                >
+                  {candidate.name}
+                </Button>
+              )}
+            />
+          ))
+          : (
+              <Controller
+                name="candidateIds"
+                control={control}
+                defaultValue={question.candidates.map(({ id }) => id)}
+                render={({ field: { value, onChange } }) => (
+                  <Reorder.Group
+                    values={value}
+                    onReorder={(v) => {
+                      setValue('type', 'PreferentialVote');
+                      onChange(v);
+                      onSubmit();
+                    }}
+                    className="flex flex-col gap-2"
+                  >
+                    {value.map((id, index) => (
+                      <Reorder.Item key={id} value={id}>
+                        <span className={twMerge('btn', lastVote?.type === 'PreferentialVote' ? 'btn-accent' : 'btn-outline')}>
+                          {`${index + 1}. ${question.candidates.find(candidate => candidate.id === id)?.name}`}
+                        </span>
+                      </Reorder.Item>
+                    ))}
+                  </Reorder.Group>
+                )}
+              />
+
             )}
-            key={candidate.id}
-            onClick={() => {
-              castVote({
-                type: 'SingleVote',
-                candidateId: candidate.id,
-              });
-            }}
-          >
-            {candidate.name}
-          </Button>
-        ))}
-        <Button
-          className={twMerge(data.lastVote?.type === 'Abstain' ? 'btn-accent' : 'btn-outline')}
-          onClick={() => {
-            castVote({
-              type: 'Abstain',
-            });
-          }}
-        >
-          Abstain
-        </Button>
+        <Controller
+          name="type"
+          control={control}
+          render={({ field: { value, onChange } }) => (
+            <div className="flex justify-center">
+              <Button
```

Comment:
Same with this button, reloads the page, even in multi candidate

Reply:
fixed